kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js

```diff
@@ -30,22 +35,136 @@ grCollectionsPanel.controller('GrCollectionsPanelCtrl', [
         ctrl.error = true;
     }).catch(() => ctrl.error = true);
 
+    ctrl.selectedImages$ = selectedImages$;
 }]);
 
-grCollectionsPanel.controller('GrNodeCtrl', ['collections', function(collections) {
+grCollectionsPanel.factory('collectionsTreeState', ['$window', function($window) {
+    const localStorageKey = 'collectionsTreeState';
+    const defaultShowState = true;
+
+    function getStateFromStorage() {
+        const jsonStr = $window.localStorage.getItem(localStorageKey) || '{}';
+        return JSON.parse(jsonStr);
+    }
+
+
+    function setState(pathId, show) {
+        const newState = angular.extend({}, getStateFromStorage(), {[pathId]: show});
+        $window.localStorage.setItem(localStorageKey, JSON.stringify(newState));
+    }
+
+    function getState(pathId) {
+        const state = getStateFromStorage();
```

Comment:
I had a recollection that accessing `localStorage` could be really slow. Can't find anything to back it up so I might be wrong, and this is probably OK for now, but could be worth addressing later as still feels like a bit of a waste not to cache it.